### PR TITLE
Login

### PR DESCRIPTION
**Pull request by calavell**
As a user
So that only I can take actions in my name
I would like to log-in

### CODE REVIEW

File: spec/spec_helper.rb [hunk 1]
```diff
@@ -1,6 +1,7 @@
 ENV["RACK_ENV"] = "test"
 
-require 'setup_test_database'
+require "setup_test_database"
+require "../db_test_connection_helper"
```
Comment (ollienorman) [inline spec/spec_helper.rb]: I needed to change this to:

require_relative "../lib/db_test_connection_helper"

Otherwise it threw an error - does it still work for you guys if you change it to this? Don't know why it's being temperamental

File: spec/features/login_spec.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
+feature "Login" do
+  scenario "User can login" do
+    visit "/login"
+    User.create(username: "test_user", email: "test@test.com", password: "test")
+    fill_in "email", with: "test@test.com"
+    fill_in "password", with: "test"
+    click_button "Enter"
+    expect(page).to have_content "Welcome"
```
Comment (ollienorman) [inline spec/features/login_spec.rb:8]: Feel maybe this should welcome the user by name - otherwise could potentially have logged in the wrong user and the test wouldn't know?

File: spec/units/user_spec.rb [hunk 3]
```diff
@@ -12,4 +12,12 @@
...
     end
   end
+  describe ".authentificate" do
+    it "checks the user exists" do
+      User.create(username: "test_user", email: "test@test.com", password: "test_password")
+      auth = User.authentificate(email: "test@test.com", password: "test_password")
+      p auth
+      expect(auth.email).to eq "test@test.com"
+    end
+  end
```
Comment (Will-Helliwell) [inline spec/units/user_spec.rb:27]: You could add another test to give yourself some reassurance that the method returns nil. 

You can also remove the p statement, and perhaps make the 'auth' variable more descriptive so that it is clearer (perhaps 'return_value'?)

File: views/login.erb [hunk 4]
```diff
@@ -0,0 +1,20 @@
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>Document</title>
+</head>
+<body>
+    <h1>Login</h1>
+    <% if flash[:email_error]  %>
```
Comment (ollienorman) [inline views/login.erb]: I'm getting some strange line-ender characters that VSCode is unhappy about in this file...not sure what they are or if it's only me that's getting them!

**Review by Will-Helliwell — COMMENTED** [review]
Really good guys! The Flash stuff is all new to me, never would have been able to implement that myself. It's very clever

File: spec/units/user_spec.rb (hunk 3 above)
Comment (calavell) [reply spec/units/user_spec.rb:27]: new test added

File: spec/spec_helper.rb (hunk 1 above)
Comment (calavell) [reply spec/spec_helper.rb]: same
